roles/openshift_autoheal/templates/objects.yml.j2

```diff
@@ -0,0 +1,118 @@
+apiVersion: v1
+kind: List
+items:
+
+- apiVersion: v1
+  kind: Namespace
+  metadata:
+    name: {{ openshift_autoheal_namespace }}
+
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+
+- apiVersion: v1
+  kind: Role
+  metadata:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  rules:
+  - apiGroups:
+    - ""
+    resources:
+    - secrets
+    resourceNames:
+    - {{ openshift_autoheal_name }}-awx-credentials
+    - {{ openshift_autoheal_name }}-awx-tls
+    verbs:
+    - get
+
+- apiVersion: v1
+  kind: RoleBinding
+  metadata:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  roleRef:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}
+  subjects:
+  - kind: ServiceAccount
+    name: {{ openshift_autoheal_name }}
+  userNames:
+  - system:serviceaccount:{{ openshift_autoheal_namespace }}:{{ openshift_autoheal_name }}
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}-awx-credentials
+  data:
+    username: {{ openshift_autoheal_awx_user | b64encode }}
+    password: {{ openshift_autoheal_awx_password | b64encode }}
+
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: {{ openshift_autoheal_namespace }}
+    name: {{ openshift_autoheal_name }}-awx-tls
+  data:
+    ca.crt: {{ openshift_autoheal_awx_ca | b64encode }}
+
+- apiVersion: v1
+  kind: ConfigMap
```

Comment:
Note that in this case there are multiple files that need to be merged into one, because the auto-heal service only supports one YAML file. What `oc_configmap` can do is put multiple files inside one secret. That does not solve this problem.

However, I have recently changed the auto-heal service so that it can load multiple configuration files from a directory. Assuming that `oc_configmap` can read multiple files without explicitly enumerating them it could work. I mean something like this:

```yaml
oc_configmap:
  ...
  from_file:
  - rules/*.yml
```

I think that isn't possible right now, but I will check.

Reply:
So the way we solved the configuration problem with Alertmanager is to store the entire contents of the config file [as an Ansible variable](https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_cluster_monitoring_operator/defaults/main.yml#L9) which is then consumed by the installer and turned into a secret. That way you can maintain a no-op default and allow the cluster operators (us) to maintain a complete config in an external/private CM system.

I think this configuration system might be simplified if the entire autoheal.yaml (including all AWX connectivity info and rules) were encoded into a single secret file and mounted in for the process to read. Then we can supply an "example" in the default autoheal config variable, and admins (like ops) can easily override.